test/units/testsuite-60.sh

```diff
@@ -53,6 +53,44 @@ EOF
     }
 }
 
+test_issue_23796() {
+    local mount_path mount_mytmpfs
+
+    mount_path="$(command -v mount 2>/dev/null)"
+    mount_mytmpfs="${mount_path/\/bin/\/sbin}.mytmpfs"
+    cat >"$mount_mytmpfs" <<EOF
+#!/bin/bash
+sleep ".\$RANDOM"
+exec -- $mount_path -t tmpfs tmpfs "\$2"
+EOF
+    chmod +x "$mount_mytmpfs"
+
+    mkdir -p /run/systemd/system
+    cat >/run/systemd/system/tmp-hoge.mount <<EOF
+[Mount]
+What=mytmpfs
+Where=/tmp/hoge
+Type=mytmpfs
+EOF
+
+    # shellcheck disable=SC2064
+    trap "rm -f /run/systemd/system/tmp-hoge.mount '$mount_mytmpfs" RETURN
```

Comment:
Whoops, I missed `'` in my previous comment, sorry about that:

```suggestion
    trap "rm -f /run/systemd/system/tmp-hoge.mount '$mount_mytmpfs'" RETURN
```

Also, I had to add a sleep between `systemctl start` and `systemctl daemon-reexec` below, to reproduce the issue reliably after reverting the first commit from this PR, i.e. the final patch to make it all work was

```diff
diff --git a/test/units/testsuite-60.sh b/test/units/testsuite-60.sh
index c282e9ac68..ebb9ae9936 100755
--- a/test/units/testsuite-60.sh
+++ b/test/units/testsuite-60.sh
@@ -74,10 +74,11 @@ Type=mytmpfs
 EOF
 
     # shellcheck disable=SC2064
-    trap "rm -f /run/systemd/system/tmp-hoge.mount '$mount_mytmpfs" RETURN
+    trap "rm -f /run/systemd/system/tmp-hoge.mount '$mount_mytmpfs'" RETURN
 
     for ((i = 0; i < 10; i++)); do
         systemctl --no-block start tmp-hoge.mount
+        sleep ".$RANDOM"
         systemctl daemon-reexec
 
         sleep 1
```